xbmc/filesystem/SmartPlaylistDirectory.cpp

```diff
@@ -233,6 +233,12 @@ namespace XFILE
         if (!videoUrl.FromString(baseDir))
           return false;
 
+        // adjust the group in case we're retrieving a grouped playlist
+        // based on artists. This is needed because the video library
+        // is using the actorslink table for artists.
+        if (isGrouped && group == "artists")
+          mvidPlaylist.SetGroup("actors");
```

Comment:
In `CSmartPlaylistDirectory::GetDirectory()` (i.e. the method this change is in) there's a local `group` variable which you check against `"artists"` but then you only change the group value in the XSP to `"actors"` and not the cached group value in the local `group` variable which is still `"artists"`.